File: Simple Linear Regression.py
# Simple Linear Regression Model

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
ds = pd.read_csv("Salary_Data.csv")
X = ds.iloc[:, :-1].values # [Rows, Columns] : = take all
Y = ds.iloc[:, 1].values

# Splitting the dataset into a training set and test set
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 1/3, random_state = 0)

# Fitting Simple Linear Regression to the training set
from sklearn.linear_model import LinearRegression
lr = LinearRegression()
lr.fit(X_train, Y_train)

# Predict the test set results
y_pred = lr.predict(X_test)

# Visulizing the training set results
plt.scatter(X_train, Y_train, color = 'red')
plt.plot(X_train, lr.predict(X_train), color = 'purple')
plt.title('Salery vs. Experience')
plt.xlabel('Years of experience')
plt.ylabel('Salery')
plt.show()


# Visulizing the test set results
plt.scatter(X_test, Y_test, color = 'red')
plt.plot(X_train, lr.predict(X_train), color = 'purple')
plt.title('Salery vs. Experience')
plt.xlabel('Years of experience')
plt.ylabel('Salery')
plt.show()

File: Simple Linear Regression.R
# Data Preprocessing

# Importing the dataset
ds = read.csv('Salary_Data.csv')

# ds = ds[, 2:3] # If needing to select specific indexes


# Splitting the dataset into a training set and test set
# install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(ds$Salary, SplitRatio = 2/3)
trs = subset(ds, split == TRUE) # Training Set
tes = subset(ds, split == FALSE) # Test Set

# Fitting Simple Linear Regression to the training set
lr = lm(formula = Salary ~ YearsExperience, data = trs)

# Predicting the test set restults
y_pred = predict(lr, newdata = tes)

# Visualising the Training set results
# install.packages('ggplot2')
# library(ggplot2)
ggplot() +
  geom_point(aes(x = trs$YearsExperience, y = trs$Salary), colour = 'red') +
  geom_line(aes(x = trs$YearsExperience, y = predict(lr, newdata = trs)), colour = 'purple') +
  ggtitle('Salary vs Experience') +
  xlab('Years of Experience') +
  ylab('Salary')
 
# Visualising the Test set results
ggplot() +
  geom_point(aes(x = tes$YearsExperience, y = tes$Salary), colour = 'red') +
  geom_line(aes(x = trs$YearsExperience, y = predict(lr, newdata = trs)), colour = 'purple') +
  ggtitle('Salary vs Experience') +
  xlab('Years of Experience') +
  ylab('Salary')


  